### ISSUE TITLE
Not list all items, and about sorting

### ISSUE THREAD

**Issue by patw0929** [gitpoint/git-point#95]
In list pages (e.g., search result list, user's repository list...etc), it seems like we don't fetch all the result but only page `1`. Maybe we can add the `per_page` parameter, but it could be a large dataset, so we might think the "fetch more" behavior in action.
Another thing is about sorting, in repo list page, the default order is sorting by full_name. Should it be same as web?

**Comment by housseindjirdeh** [comment]
Yep this is something I've thought about but decided for simplicity to stick one page for the first release. It definitely makes sense to actually show more and more of the results.

**Comment by housseindjirdeh** [comment]
If I'm not mistaken, 'FlatList' utilizes some sort of scrollLoading to allow loading more content as the user reaches close to the bottom. Definitely something we could try for each FlatList in the app that is rendered with a per-page type endpoint. I also noticed that sorting in the repo list page isn't the same as the web - if it's something to align them I'm definitely all for it. However I wouldn't struggle too much there if it's not as simple as it sounds (will also make sense to split that into a separate ticket).

**Comment by patw0929** [comment]
I'll try to implement this, about the "load more" feature.

![](http://g.recordit.co/CeWljWTcx9.gif)

Please let me know if you have any suggestion. :smile:

**Comment by housseindjirdeh** [comment]
This is great @patw0929, I'm loving this <3

The only thing I can think of is how we can expand this to all the `FlatLists` in the application and I hope it's not too much work which requires refactoring each `FlatList's` actions Redux pattern - but this is an awesome start and happy to have just search as a PR before we move on to other screens
